src/sink/rest.rs

```diff
@@ -155,7 +155,7 @@ impl Rest {
         //.map_err(|e| e.0)?;
 
         Ok(Event {
-            id,
+            id, // TODO only eid should be preserved for this?
```

Comment:
the sink should probably include it's own ID for GD/CB logic, but id's are now multi dimensional so it can perserve the id along with adding a new one.